internal/generate/packagemanifest/packagemanifest.go

```diff
@@ -56,91 +79,58 @@ type Generator struct {
 	// generated PackageManifest. If true, ChannelName will be the PackageManifest's default channel.
 	// Setting this field is only necessary when more than one channel exists.
 	IsDefaultChannel bool
-
-	// Func that returns a base PackageManifest.
-	getBase getBaseFunc
-	// Func that returns the writer the generated PackageManifest's bytes are written to.
-	getWriter func() (io.Writer, error)
-}
-
-// Type of Generator.getBase.
-type getBaseFunc func() (*apimanifests.PackageManifest, error)
-
-// Option is a function that modifies a Generator.
-type Option func(*Generator) error
-
-// WithBase sets a Generator's base PackageManifest to one that either exists or a default.
-func WithBase(inputDir string) Option {
-	return func(g *Generator) error {
-		g.getBase = g.makeBaseGetter(inputDir)
-		return nil
-	}
-}
-
-// WithWriter sets a Generator's writer to w.
-func WithWriter(w io.Writer) Option {
-	return func(g *Generator) error {
-		g.getWriter = func() (io.Writer, error) {
-			return w, nil
-		}
-		return nil
-	}
-}
-
-// WithFileWriter sets a Generator's writer to a PackageManifest file under <dir>.
-func WithFileWriter(dir string) Option {
-	return func(g *Generator) (err error) {
-		g.getWriter = func() (io.Writer, error) {
-			return genutil.Open(dir, makePkgManFileName(g.OperatorName))
-		}
-		return nil
-	}
 }
 
 // Generate configures the Generator with opts then runs it.
-func (g *Generator) Generate(opts ...Option) error {
-	for _, opt := range opts {
-		if err := opt(g); err != nil {
-			return err
-		}
+func (g PackageManifestGenerator) Generate(operatorName, version, outputDir string, opts ...*PkgOptions) error {
+	if operatorName == "" {
+		return ErrNoOpName
+	} else if version == "" {
+		return ErrNoVersion
+	} else if outputDir == "" {
+		return ErrNoOutputDir
```

Comment:
Nit: move these into their own conditions
```suggestion
	if operatorName == "" {
		return ErrNoOpName
	}
	if version == "" {
		return ErrNoVersion
	}
	if outputDir == "" {
		return ErrNoOutputDir
```